protocol-units/da-sequencer/node/src/batch.rs

```diff
@@ -1,24 +1,62 @@
 use crate::error::DaSequencerError;
 use aptos_crypto::ed25519::{Ed25519PublicKey, Ed25519Signature};
+use core::convert::TryFrom;
 
 #[derive(Debug)]
 pub struct RawData {
-	pub bytes: Vec<u8>,
+	pub data: Vec<u8>,
 }
 
 #[derive(Debug)]
 pub struct FullnodeTx {}
 
 #[derive(Debug)]
 pub struct DaBatch<Data> {
-	data: Data,
-	signature: Ed25519Signature,
-	signer: Ed25519PublicKey,
+	pub data: Data,
+	pub signature: Ed25519Signature,
+	pub signer: Ed25519PublicKey,
+	pub timestamp: u64,
+}
+
+impl DaBatch<RawData> {
+	pub fn now(signer: Ed25519PublicKey, signature: Ed25519Signature, data: Vec<u8>) -> Self {
+		let timestamp = chrono::Utc::now().timestamp_micros() as u64;
+		DaBatch { data: RawData { data }, signature, signer, timestamp }
+	}
 }
 
 /// Batch write blobs.
-fn validate_batch(
+pub fn validate_batch(
 	new_batch: DaBatch<RawData>,
 ) -> std::result::Result<DaBatch<FullnodeTx>, DaSequencerError> {
 	todo!()
 }
+
+pub fn serialize_node_batch(
```

Comment:
If I understand the intention correctly, can we update to:
```suggestion
pub fn serialize_full_node_batch(
```